js/oceanex.js

```diff
@@ -0,0 +1,323 @@
+        let result = {
+            'info': data,
+            'id': data['id'],
+            'symbol': data['market'],
+            'type': data['ord_type'],
+            'side': data['side'],
+            'status': data['state'],
+        };
+        return result;
+    }
+
+    async fetchOrders (symbol = undefined, since = undefined, limit = undefined, params = {}) {
+        let response = await this.privateGetOrders (this.extend (params));
+        const data = response['data'];
+        return this.parseOrders (data);
+    }
+
+    parseOrder (order, market = undefined) {
+        let result = {
+            'info': order,
+            'id': order['id'],
+            'timestamp': order['created_on'],
+            'datetime': order['created_at'],
+            'lastTradeTimestamp': undefined,
+            'symbol': order['market'],
+            'type': order['ord_type'],
+            'side': order['side'],
+            'price': this.safeFloat (order, 'price'),
+            'average': this.safeFloat (order, 'avg_price'),
+            'amount': this.safeFloat (order, 'volume'),
+            'remaining': this.safeFloat (order, 'remaining_volume'),
+            'filled': this.safeFloat (order, 'executed_volume'),
+            'status': order['state'],
+            'fee': undefined,
+        };
+        return result;
+    }
+
+    async createMultiOrders (params = {}) {
+        let response = await this.privatePostOrdersMulti (this.extend (params));
+        const data = response['data'];
+        return this.parseOrders (data);
+    }
+
+    async filterOrders (params = {}) {
+        let response = await this.privateGetOrdersFilter (this.extend (params));
+        const data = response['data'];
+        return this.parseOrders (data);
+    }
+
+    async cancelOrder (id, symbol = undefined, params = {}) {
+        let response = await this.privatePostOrderDelete (this.extend (params));
+        const data = response['data'];
+        return this.parseOrder (data);
+    }
+
+    async cancelMulitOrders (params = {}) {
+        let response = await this.privatePostOrderDeleteMulti (this.extend (params));
+        const data = response['data'];
+        return this.parseOrders (data);
+    }
+
+    async cancelAllOrders (symbol = undefined, params = {}) {
+        let response = await this.privatePostOrdersClear (this.extend (params));
+        const data = response['data'];
+        return this.parseOrders (data);
+    }
+
+    sign (path, api = 'public', method = 'GET', params = {}, headers = undefined, body = undefined) {
+        let url = this.urls['api'] + this.implodeParams (path, params);
+        let query = this.omit (params, this.extractParams (path));
+        if (Object.keys (query).length) {
+            url += '?' + this.urlencode (query);
+        }
+        headers = { 'Content-Type': 'application/json' };
+        return { 'url': url, 'method': method, 'body': body, 'headers': headers };
+    }
```

Comment:
Looks like the implementation of the `sign` method is missing a `public` and `private` section (all requests are passed through this method, not just the public ones), so, we need to rework it and make it compatible with both the authenticated and non-authenticated sections of the API.

See how this method is implemented here, namely, check the public and private sections of it:

- https://github.com/ccxt/ccxt/blob/master/js/upbit.js#L1458